ibis/tests/expr/test_table.py

```diff
@@ -1812,3 +1819,9 @@ def test_invalid_keep_distinct():
     t = ibis.table(dict(a="int", b="string"), name="t")
     with pytest.raises(com.IbisError, match="Invalid value for `keep`:"):
         t.distinct(on="a", keep="invalid")
+
+
+def test_invalid_distinct_empty_key():
+    t = ibis.table(dict(a="int", b="string"), name="t")
+    with pytest.raises(ValueError):
+        t.distinct(on="c", keep="first")
```

Comment:
I think we should still raise a more specific error message in `distinct(on="does_not_exist")`. Erroring in the group by that is the implementation of `distinct(on...)` leaks implementation details and may leave users wondering why a group by is involved.

Reply:
This can be done in a follow-up for sure!

Reply:
I'll try to do it in a follow-up since I've to figure out how to do it.